plugin/grpc/grpcstats/client_handler.go

```diff
@@ -160,12 +161,14 @@ func (h *ClientStatsHandler) handleRPCEnd(ctx context.Context, s *stats.End) {
 	m = append(m, RPCClientRoundTripLatency.M(float64(elapsedTime)/float64(time.Millisecond)))
 
 	if s.Error != nil {
-		errorCode := s.Error.Error()
-		newTagMap, err := tag.NewMap(ctx,
-			tag.Upsert(keyOpStatus, errorCode),
-		)
-		if err == nil {
-			ctx = tag.NewContext(ctx, newTagMap)
+		s, ok := status.FromError(s.Error)
+		if ok {
```

Comment:
Could we keep the old path for cases where FromError returns ok: false?
```go
var errorCode string
if s, ok := status.FromError(s.Error); ok {
   errorCode = s.Code().String()
} else {
   errorCode = s.Error().Error()
}
newTagMap, err := tag.NewMap(ctx, tag.Upsert(keyStatus, errorCode))
if err != nil {
   ctx = tag.NewContext(ctx, newTagMap)
}
```

Reply:
We don't do it because a free formatted message is not a dimension.

Reply:
But we did use it before though, that's a fallback.